### ISSUE TITLE
xenial ppc64el image shows odd size

### ISSUE THREAD

**Issue by smoser** [lxc/lxd#2324]
I'm not really sure what could have done this, but the xenial image on my ppc64el host shows much larger than trusty or yakkety.  'du' shows that xenial *is* larger than yakkety from inside, but not so much to account for the drastic difference.  Additionally, information from http://cloud-images.ubuntu.com/daily/server/xenial/20160826/ shows 131M for the 20160826 xenial image.

    $ lxc image list
    +---------+--------------+--------+---------------------------------------------+---------+----------+------------------------------+
    |  ALIAS  | FINGERPRINT  | PUBLIC |                 DESCRIPTION                 |  ARCH   |   SIZE   |         UPLOAD DATE          |
    +---------+--------------+--------+---------------------------------------------+---------+----------+------------------------------+
    | trusty  | 0c515b51eaf8 | no     | ubuntu 14.04 LTS ppc64el (daily) (20160826) | ppc64le | 117.84MB | Aug 29, 2016 at 2:52am (UTC) |
    +---------+--------------+--------+---------------------------------------------+---------+----------+------------------------------+
    | xenial  | 302173a894c1 | no     | ubuntu 16.04 LTS ppc64el (daily) (20160826) | ppc64le | 313.89MB | Aug 29, 2016 at 2:53pm (UTC) |
    +---------+--------------+--------+---------------------------------------------+---------+----------+------------------------------+
    | yakkety | 948867a651f4 | no     | ubuntu 16.10 ppc64el (daily) (20160828)     | ppc64le | 150.59MB | Aug 29, 2016 at 2:50am (UTC) |
    +---------+--------------+--------+---------------------------------------------+---------+----------+------------------------------+
    $ lxc image info 302173a894c1
    Fingerprint: 302173a894c13e6adc1e9808cc88a8ca2e89b1abb7535f54ff62ae5cdfb0deaa
    Size: 313.89MB
    Architecture: ppc64le
    Public: no
    Timestamps:
        Created: 2016/08/26 00:00 UTC
        Uploaded: 2016/08/29 14:53 UTC
        Expires: 2021/04/21 00:00 UTC
    Properties:
        os: ubuntu
        release: xenial
        serial: 20160826
        version: 16.04
        aliases: 16.04,default,lts,x,xenial
        architecture: ppc64el
        description: ubuntu 16.04 LTS ppc64el (daily) (20160826)
        label: daily
    Aliases:
        - xenial
    Auto update: enabled
    Source:
        Server: https://cloud-images.ubuntu.com/daily
        Protocol: simplestreams
        Alias: xenial
    $ lxc image info 948867a651f4
    Fingerprint: 948867a651f43335721bc85743316485951cd3b4e918446429c49e6b9157328c
    Size: 150.59MB
    Architecture: ppc64le
    Public: no
    Timestamps:
        Created: 2016/08/28 00:00 UTC
        Uploaded: 2016/08/29 02:50 UTC
        Expires: 2017/07/20 00:00 UTC
    Properties:
        description: ubuntu 16.10 ppc64el (daily) (20160828)
        label: daily
        os: ubuntu
        release: yakkety
        serial: 20160828
        version: 16.10
        aliases: 16.10,devel,y,yakkety
        architecture: ppc64el
    Aliases:
        - yakkety
    Auto update: enabled
    Source:
        Server: https://cloud-images.ubuntu.com/daily
        Protocol: simplestreams
        Alias: yakkety

    $ lxc launch yakkety y1
    Creating y1
    Starting y1
    $ lxc launch xenial x1

    $ lxc exec y1 -- du -hs --one-file-system /
    707M  /
    $ lxc exec x1 -- du -hs --one-file-system /
    788M  /

    $ lxc exec y1 -- cat /etc/cloud/build.info
    build_name: server
    serial: 20160828
    $ lxc exec x1 -- cat /etc/cloud/build.info
    build_name: server
    serial: 20160826

**Comment by smoser** [comment]
$ sudo sh -c 'cd /var/lib/lxd/images/; for x in *.rootfs; do echo == $x ==; ls -l $x; tar -xf $x --to-stdout etc/issue etc/cloud/build.info ; done'
    == 0c515b51eaf833f34cdbe22907d9c5786842332a43092990fc6e52e083a91140.rootfs ==
    -rw-r--r-- 1 root root 123565492 Aug 29 02:52 0c515b51eaf833f34cdbe22907d9c5786842332a43092990fc6e52e083a91140.rootfs
    build_name: server
    serial: 20160826
    Ubuntu 14.04.5 LTS \n \l
    
    == 302173a894c13e6adc1e9808cc88a8ca2e89b1abb7535f54ff62ae5cdfb0deaa.rootfs ==
    -rw-r--r-- 1 root root 145326840 Aug 29 14:53 302173a894c13e6adc1e9808cc88a8ca2e89b1abb7535f54ff62ae5cdfb0deaa.rootfs
    Ubuntu 16.04.1 LTS \n \l
    
    build_name: server
    serial: 20160826
    == 948867a651f43335721bc85743316485951cd3b4e918446429c49e6b9157328c.rootfs ==
    -rw-r--r-- 1 root root 157908992 Aug 29 02:50 948867a651f43335721bc85743316485951cd3b4e918446429c49e6b9157328c.rootfs
    tar: This does not look like a tar archive
    tar: Skipping to next header
    tar: etc/issue: Not found in archive
    tar: etc/cloud/build.info: Not found in archive
    tar: Exiting with failure status due to previous errors
    
    $ sudo mount-image-callback /var/lib/lxd/images/948867a651f43335721bc85743316485951cd3b4e918446429c49e6b9157328c.rootfs --cd cat etc/issue etc/cloud/build.info
    Ubuntu Yakkety Yak (development branch) \n \l

    build_name: server
    serial: 20160828

**Comment by smoser** [comment]
On a side note, it would be nice to actually be able to ask lxd what size in bytes is, rather than having it tell you the size  rounded to some unit.

**Comment by stgraber** [comment]
We had a couple of bugs related to our simplestreams client code getting a bit confused when a series had both a .tar.xz and a .squashfs image available. Those bugs have been fixed in 2.0.4 or 2.1.

I just confirmed on both a 2.0.4 and 2.1 system that the size reported for the remote image "lxc image list ubuntu:" and the image size after a "lxc image copy ubuntu:xyz local:" are both correct.

Closing this.
